src/Workspaces/Core/Portable/Options/DocumentOptionSet.cs

```diff
@@ -26,9 +26,12 @@ internal DocumentOptionSet(OptionSet backingOptionSet, string language)
             _language = language;
         }
 
-        public override object? GetOption(OptionKey optionKey)
+        private protected override object? GetOptionCore(OptionKey optionKey)
             => _backingOptionSet.GetOption(optionKey);
 
+        public new object? GetOption(OptionKey optionKey)
+            => base.GetOption(optionKey);
```

Comment:
This may warrant a good comment here explaining why this is here as a binary back-compat member.

Reply:
Sure, will add in a follow-up PR.